sympy/physics/mechanics/abstract_body.py

```diff
@@ -0,0 +1,123 @@
+from abc import ABC, abstractmethod
+
+from sympy.core.backend import Symbol, sympify
+from sympy.physics.vector import Point
+
+__all__ = ['_Body']
+
+
+class _Body(ABC):
+    """Abstract class for body type objects."""
+    def __init__(self, name, masscenter=None, frame=None, mass=None):
+        # Note: If frame=None, no auto-generated frame is created, because a
+        # Particle does not need to have a frame by default.
+        if not isinstance(name, str):
+            raise TypeError('Supply a valid name.')
```

Comment:
This isn't the most useful error message. Can we make it clearer?
```suggestion
            raise TypeError(f'Supplied name {name} is invalid, must be a {str}.')
```

Reply:
Similar error messaging is used for `Joint`, `ReferenceFrame` and `Point`

Reply:
Okay, ignore for now. Could be improved in the future.